rust/src/lib/nm/nm_dbus/gen_conf/route_rule.rs

```diff
@@ -1,3 +1,54 @@
 // SPDX-License-Identifier: Apache-2.0
 
-// TODO
+use std::collections::HashMap;
+
+use super::super::NmIpRouteRule;
+
+const DEFAULT_IPV4: &str = "0.0.0.0";
+const DEFAULT_IPV6: &str = "::";
+const DEFAULT_IP_PREFIX: u8 = 0;
+const DEFAULT_ROUTE_TABLE: u32 = 254;
+const AF_INET: i32 = 2;
+
+impl NmIpRouteRule {
+    pub(crate) fn to_keyfile(&self) -> HashMap<String, String> {
+        let mut ret = HashMap::new();
+        if let Some(priority) = self.priority.as_ref() {
+            let prio_str = format!("priority {}", priority);
+            let to_str = match (self.to.as_ref(), self.to_len.as_ref()) {
+                (Some(t), Some(t_len)) => format!("to {}/{}", t, t_len),
+                (Some(t), None) => format!("to {}", t),
+                _ => "".to_string(),
+            };
+            let from_str = match (self.from.as_ref(), self.from_len.as_ref()) {
+                (Some(f), Some(f_len)) => format!("from {}/{}", f, f_len),
+                (Some(f), None) => format!("from {}", f),
+                _ => {
+                    // we need to check if ip_to is empty or not
+                    if to_str.is_empty() {
+                        format!(
+                            "from {}/{}",
+                            if self.family.unwrap_or_default() == AF_INET {
+                                DEFAULT_IPV4
+                            } else {
+                                DEFAULT_IPV6
+                            },
+                            DEFAULT_IP_PREFIX
```

Comment:
Just FYI. Nmstate will raise error if both `ip-to` and `ip-from` is empty.

Reply:
In that case, we don't need to handle that case. Let me drop it. Thanks!